src/bencher/args.rs

```diff
@@ -0,0 +1,206 @@
+    )]
+    pub(crate) path: String,
+
+    #[command(subcommand)]
+    pub(crate) command: BencherCommands,
+}
+
+#[derive(Subcommand, Clone)]
+pub(crate) enum BencherCommands {
+    Db(BenchmarkDbArgs),
+}
+
+#[derive(Args, Clone)]
+#[command(about = "Benchmark a SlateDB database.")]
+
+pub(crate) struct DbArgs {
+    #[arg(
+        long,
+        help = "Whether to disable the write-ahead log.",
+        default_value_t = false
+    )]
+    pub(crate) disable_wal: bool,
+
+    #[arg(
+        long,
+        help = "The interval in milliseconds to flush the write-ahead log."
+    )]
+    pub(crate) flush_ms: Option<u32>,
+
+    #[arg(long, help = "The size in bytes of the L0 SSTables.")]
+    pub(crate) l0_sst_size_bytes: Option<usize>,
+
+    #[arg(long, help = "The size in bytes of the block cache.")]
+    pub(crate) block_cache_size: Option<usize>,
+
+    #[arg(
+        long,
+        help = "The path where object store cache part files are stored."
+    )]
+    pub(crate) object_cache_path: Option<String>,
+
+    #[arg(long, help = "The size in bytes of the object store cache part files.")]
+    pub(crate) object_cache_part_size: Option<usize>,
+}
+
+impl DbArgs {
+    /// Returns a `DbOptions` struct based on DbArgs's arguments.
+    #[allow(dead_code)]
+    pub(crate) fn config(&self) -> DbOptions {
+        let mut db_options = DbOptions::default();
+        db_options.wal_enabled = !self.disable_wal;
+        db_options.flush_interval = self
+            .flush_ms
+            .map(|i| Duration::from_millis(i as u64))
+            .unwrap_or(db_options.flush_interval);
+        db_options.l0_sst_size_bytes = self
+            .l0_sst_size_bytes
+            .unwrap_or(db_options.l0_sst_size_bytes);
+        db_options.block_cache = self.block_cache_size.map(|size| {
+            Arc::new(MokaCache::new_with_opts(MokaCacheOptions {
+                max_capacity: size as u64,
+                ..Default::default()
+            })) as Arc<dyn DbCache>
+        });
+        db_options.object_store_cache_options = self
+            .object_cache_path
+            .clone()
+            .map(|path| ObjectStoreCacheOptions {
+                root_folder: Some(PathBuf::from(path)),
+                ..Default::default()
+            })
+            .unwrap_or(db_options.object_store_cache_options);
+        db_options.object_store_cache_options.part_size_bytes = self
+            .object_cache_part_size
+            .unwrap_or(db_options.object_store_cache_options.part_size_bytes);
+        db_options
+    }
+}
+
+#[derive(Args, Clone)]
+pub(crate) struct BenchmarkDbArgs {
+    #[clap(flatten)]
+    pub(crate) db_args: DbArgs,
+
+    #[arg(long, help = "The duration in seconds to run the benchmark for.")]
+    pub(crate) duration: Option<u32>,
+
+    #[arg(
+        long,
+        help = "The key distribution to use.",
+        default_value_t = KeyDistribution::Random
+    )]
+    pub(crate) key_distribution: KeyDistribution,
```

Comment:
nit: I'd prefer to call this a "key generator" (I realize that I also called it a distribution in db_bench 😄). I can imagine us supporting other modes of generating keys that are not sourced from a distribution. For example, if we want to test cache effectiveness we might want to replay a trace of a real workload.

Reply:
bfaa61e